sound/soc/sof/pcm.c

```diff
@@ -318,6 +318,10 @@ static int sof_pcm_trigger(struct snd_pcm_substream *substream, int cmd)
 	snd_sof_pcm_platform_trigger(sdev, substream, cmd);
 
 	/* send IPC to the DSP */
+	/*
+	 * the trigger action is in non-atomic environment, which
+	 * is set in topology parsing.
+	 */
```

Comment:
@libinyang the comment here looks very ambiguous. A better place would be just a few lines above where we return 0 for BE:
/* nothing todo for BE */

We could call out that the trigger is purely for FE's and that the non-atomic field for FE's is explicitly set to true.

Reply:
@ranj063 
Actually, it is not related to FE/BE trigger. Both FE/BE trigger will be in non-atomic environment in our case.

Reply:
@plbossart 
Do you mean I should add the comment about why FE is non-atomic will cause the trigger non-atomic? I think this is a alsa/asoc level general question. I'm afraid that the maintainers may be not interested in this? For the comments, I believe we just need to explain why our code is OK. We don't need to explain what other drivers flow. What do you think?
From Takashi's comments, my understanding is Takashi just want to  make sure trigger is in non-atomic environment, and when it is set. Maybe I misunderstand Takashi's meaning. If you think we need to explain why setting non-atomic in FE can make sure the trigger is in non-atomic environment, I will add the details. But the comments may be a little complicated and boring.

Reply:
I just want to clarify what assumptions are made for FE and BE and check that this is aligned with the previous solutions. not sure what's so complicated?

Reply:
@plbossart 
I think I got you now. Do you want the clarify in the comments of the code or the description header of the patch?

Today, I will work on hdmi audio patch firstly. Maybe the comments are ready tomorrow.